packages/ag-charts-community/src/chart/series/cartesian/cartesianSeries.ts

```diff
@@ -841,99 +831,77 @@ export abstract class CartesianSeries<
-
-        // Find the max visible distance between adjacent points in both directions
-        let maxVisibleWidth = 0;
-        let maxVisibleHeight = 0;
-
-        for (let i = 1; i < Math.max(minVisibleRectXs.length, minVisibleRectYs.length); i++) {
-            const x1 = minVisibleRectXs[i];
-            const x2 = minVisibleRectXs[i - 1];
-            const y1 = minVisibleRectYs[i];
-            const y2 = minVisibleRectYs[i - 1];
-
-            if (x1 != null && x2 != null) {
-                maxVisibleWidth = Math.max(maxVisibleWidth, x1 - x2);
-            }
+        return visibleItems;
+    }
 
-            if (y1 != null && y2 != null) {
-                maxVisibleHeight = Math.max(maxVisibleHeight, y1 - y2);
+    protected countVisibleItems(
+        xKey: string,
+        yKey: string,
+        xVisibleRange: [number, number],
+        yVisibleRange: [number, number],
+        minVisibleItems: number
+    ): number {
+        const { dataModel, processedData } = this;
+        if (!dataModel || !processedData) return Infinity;
+
+        const xValues = this.keysOrValues(xKey);
+        const yValues = dataModel.resolveColumnById(this, yKey, processedData);
+
+        // The provided visible ranges `[xy]VisibleRange` are relative to the unzoomed axis ranges `[xy]Axis.range`.
+        // `[xy]Axis.visibleRange` are relative to the zoomed scale ranges `[xy]Axis.scale.range`.
+        // So we need to scale the provided visible ranges relative to the zoomed scale range to find the min and max
+        // position in pixels that the visible range ratio covers.
+        const xAxis = this.axes[ChartAxisDirection.X]!;
+        const yAxis = this.axes[ChartAxisDirection.Y]!;
+
+        const xRangeSize = xAxis.range[1] - xAxis.range[0];
+        const xVisibleRangeSize = xAxis.visibleRange[1] - xAxis.visibleRange[0];
+        const yRangeSize = yAxis.range[0] - yAxis.range[1];
+        const yVisibleRangeSize = yAxis.visibleRange[1] - yAxis.visibleRange[0];
+
+        const xMin = xAxis.range[0] + xRangeSize * ((xVisibleRange[0] - xAxis.visibleRange[0]) / xVisibleRangeSize);
+        const xMax = xAxis.range[0] + xRangeSize * ((xVisibleRange[1] - xAxis.visibleRange[0]) / xVisibleRangeSize);
+        const yMin = yAxis.range[0] - yRangeSize * ((yVisibleRange[1] - yAxis.visibleRange[0]) / yVisibleRangeSize);
+        const yMax = yAxis.range[0] - yRangeSize * ((yVisibleRange[0] - yAxis.visibleRange[0]) / yVisibleRangeSize);
+
+        let visibleItems = 0;
+        for (let index = 0; index < xValues.length; index++) {
+            const x = xAxis.scale.convert(xValues[index]);
+            const y = yAxis.scale.convert(yValues[index]);
+            if (x >= xMin && x <= xMax && y >= yMin && y <= yMax) {
+                visibleItems += 1;
+                if (visibleItems >= minVisibleItems) {
+                    return visibleItems;
+                }
             }
         }
```

Comment:
@jacobp100 Is there an existing method that works like this? From what I can tell, the related fns are looking for indicies, whereas I need an actual count of items.

Reply:
Jacob suggested modifying `domainForVisibleRange()` and adding a `yCoordinateRange()` method, similar to the existing `xCoordinateRange()`.

I did the latter, but I decided to keep this counting a separate method as it has a different optimisation technique (see `countExpandingSearch()`).